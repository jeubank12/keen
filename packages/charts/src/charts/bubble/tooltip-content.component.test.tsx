import React from 'react';
import { render as rtlRender } from '@testing-library/react';

import TooltipContent from './tooltip-content.component';

const render = (overProps: any = {}) => {
  const data = [
    { channel: 'Facebook', cost: 310, conversion: 352, users: 500 },
  ];
  const labelSelector = 'channel';
  const selectors = [{ color: '#487650', selector: [0] }];
  const valueKey = 'cost';
  const xDomainKey = 'users';
  const yDomainKey = 'conversion';
  const formatTooltip = {
    xKey: value => `${value} X`,
    yKey: value => `${value} Y`,
    valueKey: value => `${value} value`,
  };

  const props = {
    data,
    labelSelector,
    selectors,
    valueKey,
    xDomainKey,
    yDomainKey,
    formatTooltip,
    ...overProps,
  };

  const wrapper = rtlRender(<TooltipContent {...props} />);

  return {
    wrapper,
    props,
  };
};

test('formats tooltip value', () => {
  const {
    wrapper: { getByTestId },
  } = render();
  const tooltipContent = getByTestId('tooltip-content');

  expect(tooltipContent).toMatchSnapshot();
});
