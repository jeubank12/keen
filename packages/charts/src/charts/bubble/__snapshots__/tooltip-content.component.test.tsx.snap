// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formats tooltip value 1`] = `
<div
  data-testid="tooltip-content"
>
  <div
    class="sc-AxjAm ldjeaj"
  >
    <div
      class="sc-fzocqA fPCMGP"
    >
      <div
        class="sc-fzqKxP jHiXAe"
        color="#487650"
      />
      Facebook
    </div>
    <div
      class="sc-fzonZV ksMmQR"
    >
      <div>
        conversion
      </div>
      <div
        class="sc-fzqyOu gZubFk"
      >
        352 Y
      </div>
    </div>
    <div
      class="sc-fzonZV ksMmQR"
    >
      <div>
        users
      </div>
      <div
        class="sc-fzqyOu gZubFk"
      >
        500 X
      </div>
    </div>
    <div
      class="sc-pANHa fwrvOa"
    >
      <div>
        cost
      </div>
      <div
        class="sc-fzqyOu gZubFk"
      >
        310 value
      </div>
    </div>
  </div>
</div>
`;
